pennylane/templates/subroutines/arithmetic/temporary_and.py

```diff
@@ -190,32 +197,21 @@ def compute_decomposition(
         **Example:**
 
         >>> print(qml.TemporaryAND.compute_decomposition((0,1,2)))
-        [H(2),
-        T(2),
-        CNOT(wires=[1, 2]),
-        Adjoint(T(2)),
-        CNOT(wires=[0, 2]),
-        T(2),
-        CNOT(wires=[1, 2]),
-        Adjoint(T(2)),
-        H(2),
-        Adjoint(S(2))]
+        [(H(2) @ ((Adjoint(T(2))) @ (CNOT(wires=[1, 2])) @ T(2))) @ (CNOT(wires=[0, 2])) @ (((Adjoint(T(2))) @ (CNOT(wires=[1, 2])) @ T(2)) @ H(2)), Adjoint(S(2))]
         """
 
         list_decomp = []
 
         list_decomp.extend([X(wires[idx]) for idx in [0, 1] if control_values[idx] == 0])
 
         list_decomp += [
-            Hadamard(wires=wires[2]),
-            T(wires=wires[2]),
-            CNOT(wires=[wires[1], wires[2]]),
-            adjoint(T(wires=wires[2])),
-            CNOT(wires=[wires[0], wires[2]]),
-            T(wires=wires[2]),
-            CNOT(wires=[wires[1], wires[2]]),
-            adjoint(T(wires=wires[2])),
-            Hadamard(wires=wires[2]),
+            change_op_basis(
+                change_op_basis(T(wires=wires[2]), CNOT(wires=[wires[1], wires[2]]))
+                @ Hadamard(wires=wires[2]),
+                CNOT(wires=[wires[0], wires[2]]),
+                Hadamard(wires=wires[2])
+                @ change_op_basis(T(wires=wires[2]), CNOT(wires=[wires[1], wires[2]])),
+            ),
```

Comment:
This really gets into splitting hairs, but _teeechnically_, we could say this?

```suggestion
            change_op_basis(
                Hadamard(wires=wires[2]),
                change_op_basis(
                    change_op_basis(T(wires=wires[2]), CNOT(wires=[wires[1], wires[2]]))
                    CNOT(wires=[wires[0], wires[2]]),
                    change_op_basis(T(wires=wires[2]), CNOT(wires=[wires[1], wires[2]])),
                ),
            )
```